accepted/0000-sbom-command.md

```diff
@@ -0,0 +1,417 @@
+  "documentNamespace": "http://spdx.org/spdxdocs/hello-world-1.0.0-<uuid>",
+  "creationInfo": {
+    "created": "2023-08-04T21:41:02.071Z",
+    "creators": [
+      "Tool: pkg:npm/cli@9.8.1"
+    ]
+  },
+  "packages": [
+    {
+      "name": "hello-world",
+      "SPDXID": "SPDXRef-Package-hello-world-1.0.0",
+      "versionInfo": "1.0.0",
+      "downloadLocation": "NOASSERTION",
+      "filesAnalyzed": false,
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "npm",
+          "referenceLocator": "hello-world@1.0.0"
+        },
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "purl",
+          "referenceLocator": "pkg:npm/hello-world@1.0.0"
+        }
+      ]
+    },
+    {
+      "name": "@tsconfig/node14",
+      "SPDXID": "SPDXRef-Package-tsconfig.node14-1.0.3",
+      "versionInfo": "1.0.3",
+      "downloadLocation": "https://registry.npmjs.org/@tsconfig/node14/...",
+      "filesAnalyzed": false,
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "npm",
+          "referenceLocator": "@tsconfig/node14@1.0.3"
+        },
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "purl",
+          "referenceLocator": "pkg:npm/%40tsconfig/node14@1.0.3"
+        }
+      ],
+      "checksums": [
+        {
+          "algorithm": "SHA512",
+          "checksumValue": "cac4fc9a1762c562..."
+        }
+      ]
+    },
+    {
+      "name": "debug",
+      "SPDXID": "SPDXRef-Package-debug-4.3.4",
+      "versionInfo": "4.3.4",
+      "downloadLocation": "https://registry.npmjs.org/debug/-/debug-4.3.4.tgz",
+      "filesAnalyzed": false,
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "npm",
+          "referenceLocator": "debug@4.3.4"
+        },
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "purl",
+          "referenceLocator": "pkg:npm/debug@4.3.4"
+        }
+      ],
+      "checksums": [
+        {
+          "algorithm": "SHA512",
+          "checksumValue": "3d15851ee494dde0..."
+        }
+      ]
+    },
+    {
+      "name": "ms",
+      "SPDXID": "SPDXRef-Package-ms-2.1.2",
+      "versionInfo": "2.1.2",
+      "downloadLocation": "https://registry.npmjs.org/ms/-/ms-2.1.2.tgz",
+      "filesAnalyzed": false,
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "npm",
+          "referenceLocator": "ms@2.1.2"
+        },
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referernceType": "purl",
+          "referenceLocator": "pkg:npm/ms@2.1.2"
+        }
+      ],
+      "checksums": [
+        {
+          "algorithm": "SHA512",
+          "checksumValue": "b0690fc7e56332d9..."
+        }
+      ]
+    }
+  ]
```

Comment:
The document needs to have its relationships, it should look something similar to the following:

```suggestion
  ],
  "relationships": [
    {
      "spdxElementId": "SPDXRef-DOCUMENT",
      "relatedSpdxElement": "SPDXRef-Package-hello-world-1.0.0",
      "relationshipType": "DESCRIBES"
    },   
    {
      "spdxElementId": "SPDXRef-Package-hello-world-1.0.0",
      "relationshipType": "DEPENDS_ON",
      "relatedSpdxElement": "SPDXRef-Package-tsconfig.node14-1.0.3"
    },
    {
      "spdxElementId": "SPDXRef-Package-hello-world-1.0.0",
      "relationshipType": "DEPENDS_ON",
      "relatedSpdxElement": "SPDXRef-Package-debug-4.3.4"
    },
    {
      "spdxElementId": "SPDXRef-Package-debug-4.3.4",
      "relationshipType": "DEV_TOOL_OF",
      "relatedSpdxElement": "SPDXRef-Package-ms-2.1.2"
    }
  ]
```

Note the first relationship. It is the same data as the `documentDescribes` field above.

Reply:
@puerco thanks for all of the great feedback!

I was a little unsure about the `relationships` collection. I had that in an early draft of the RFC but then removed it after noticing that a lot of the tools I looked at weren't including it (and none of the SPDX-consuming tools I played with seemed to care if it was present). If you think it would be useful, I'm happy to put it back in (I know CycloneDX has a similar structure for capturing relationships as well).